File: tab-share-api/internal/api/models/export_models.go
package models

type ExportTab struct {
	Name       string `json:"name" validate:"required,min=1" mod:"trim"`
	FaviconUrl string `json:"favIconUrl" validate:"required,min=1" mod:"trim"`
	Url        string `json:"url" validate:"required,url" mod:"trim"`
}

type ExportRequest struct {
	Tabs []ExportTab `json:"tabs" validate:"required,dive"`
}

File: tab-share-client/src/app/tab-export/state/export.models.ts
import {ChromeTabWithId} from '../../core/state/core.models';

export type ExportState = {
  excludedTabs: ExcludedTabs;
  currentWindowTabs: ChromeTabWithId[];
  exportCode: string;
};

export type ExportTab = {
  url: string;
  name: string;
  favIconUrl?: string;
};

export type ExcludedTabs = { [key: string]: boolean };

export type ExportTabsRequest = {
  tabs: ExportTab[];
};

export type ExportTabsResponse = {
  code: string;
};
